lib/broadway_kafka/producer.ex

```diff
@@ -285,7 +306,12 @@ defmodule BroadwayKafka.Producer do
     if Acknowledger.all_drained?(state.acks) do
       {:reply, :ok, [], %{state | acks: Acknowledger.new()}}
     else
-      {:noreply, [], %{state | revoke_caller: from}}
+      {:noreply, [],
+       %{
+         state
+         | revoke_caller: from,
+           draining_after_revoke_started_at: System.monotonic_time(:second)
```

Comment:
Instead of adding a new map key, you can store all information on `revoke_caller` as a tuple, as they are used together.